test/vmtests/vmtests/test_no_change.py

```diff
@@ -58,4 +68,20 @@ def test_no_change(
     vm.start()
 
     # Wait for VM to boot by waiting for it to request an IP address from the DHCP server.
-    vm.get_vm_ip_address(timeout=30)
+    vm_ip_address = vm.get_vm_ip_address(timeout=30)
+
+    # Connect to VM using SSH.
+    ssh_known_hosts_path = test_temp_dir.joinpath("known_hosts")
+    open(ssh_known_hosts_path, "w").close()
+
+    with SshClient(vm_ip_address, key_path=ssh_private_key_path, known_hosts_path=ssh_known_hosts_path) as vm_ssh:
+        vm_ssh.run("cat /proc/cmdline").check_exit_code()
+
+        os_release_path = test_temp_dir.joinpath("os-release")
+        vm_ssh.get_file(Path("/etc/os-release"), os_release_path)
+
+        with open(os_release_path, "r") as os_release_fd:
+            os_release_text = os_release_fd.read()
+
+            assert "ID=mariner" in os_release_text
```

Comment:
Curious, why assert on being mariner 2.0?

Reply:
(this code seems to be agnostic of the target os)

Reply:
This test is currently running against the `core_efi_azl2` base image. So, asserting the image is Mariner 2.0 here is fine.

At some point, we will need to expand this test suite so that it tests against multiple Azure Linux versions. And we will need to ensure that tests can change their verification logic based on the input image.